Дока на пример правильной обработки цикла ожидания завершения процесса #402

artbear opened the pull request:
Дока на пример правильной обработки цикла ожидания завершения процесса #402

artbear committed:
Документирован пример правильной обработки цикла ожидания завершения процесса #402

artbear commented:
/cc @EvilBeaver @dmpas

dmpas commented:
https://msdn.microsoft.com/en-us/library/f8hahtxf.aspx
https://msdn.microsoft.com/en-us/library/9w4cf933.aspx

Можно обернуть пример в тэги code и example. Сам не пробовал, но в MSDN есть такое.

artbear commented:
Да, могу поправить.
Один вопрос - А умеет ли 1с-синтакс работать с этими тегами? code и example
/cc @nixel2007 @bambr1975 Как в 1с-синтакс создается синтакс-помощник?

bambr1975 commented:
да, СП в 1с-синтакс обрабатывает тег example https://github.com/xDrivenDevelopment/vsc-language-1c-bsl/blob/master/src/features/oscriptStdLib.ts#L259 - выводит его отдельным абзацем с заголовком Пример

artbear committed:
Выделил спец.теги для документации  #402

artbear commented:
@dmpas Выделил тег примеров.
Еще есть дополнения?

dmpas commented:
Думаю, в примере стоит пояснить, в чём "правильность". Или отсылку дать куда-нибудь. А то так со стороны не понять, мне кажется. А в остальном всё хорошо.

artbear committed:
описал "правильность" #402

artbear commented:
@dmpas Я упорный :)
Сделано.

dmpas commented:
У меня больше нет вопросов. @EvilBeaver  ?

EvilBeaver reviewed (commented):
Копипаст мюиз гиттера это хорошо, но не совсем точно. Речь не только про stderr, а вообще про потоки вывода. Смысл в том что читать надо оба.  Просьба переформулировать

artbear commented:
Андрей, а самому сложно потратить минуту и напрямую в моем PR поправить 2 строчки, не?
кучу времени в очередной раз убиваем

artbear commented:
Мое упорство также имеет пределы :(
который день уже копеечные изменения не можем закинуть

EvilBeaver commented:
@artbear ты сам меня приучил докапываться до мелочей, разве нет? Там суть как раз в том, что вообще про потоки речь, а не про только stderr. Я думал ты взялся это описать, а выяснилось, что получился копипаст с гиттера, в котором формулировки не совсем понятные.

В результате из этого текста можно понять, что обязательно читать нужно только stderr, а stdout и сам закроется. А это не так.

artbear committed:
Исправил формулировки правильности завершения процесса

        ///     Процесс не завершается, пока любоой из потоков (stdout, stderr) открыт для чтения.
        ///     Процесс висит и ждет, пока его освободят от текста в обоих потоках.

artbear commented:
В честь праздника сам на  гитхабе исправил.
```
///     Процесс не завершается, пока любоой из потоков (stdout, stderr) открыт для 
///     Процесс висит и ждет, пока его освободят от текста в обоих потоках.
```
@EvilBeaver Устроит?

PS поехали на Таганку пива пить?

EvilBeaver commented:
На таганку могу в будни только

bambr1975 commented on src/ScriptEngine.HostedScript/Library/ProcessContext.cs:
```diff
@@ -120,8 +120,8 @@ public void Start()
+        ///     Процесс висит и ждет, пока его освободят от текста в обоих потоках.
         ///
         /// Пока НЕ Процесс.Завершен ИЛИ Процесс.ПотокВывода.ЕстьДанные ИЛИ Процесс.ПотокОшибок.ЕстьДанные Цикл
         /// 	    Если ПериодОпросаВМиллисекундах <> 0 Тогда
```
Надо исправить на Если ПериодОпросаВМиллисекундах ```&lt;&gt;``` 0 Тогда
потому что         ```<!-- Проигнорирован некорректный комментарий XML для члена "P:ScriptEngine.HostedScript.Library.ProcessContext.HasExited" -->```
